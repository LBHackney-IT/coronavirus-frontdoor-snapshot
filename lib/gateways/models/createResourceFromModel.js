import Resource from 'lib/domain/resource';

// .filter() will ignore null or undefined values
const filterEmptyValues = item => item;

export function createResourceFromModel({ id, fields }) {
  const websites = [fields['Website 1'], fields['Website 2']].filter(
    filterEmptyValues
  );

  return new Resource({
    id: id,
    name: fields.Name,
    description: fields.Description,
    websites,
    address: fields.Address,
    postcode: fields.Postcode,
    tags: fields['Vulnerability snapshot tags'],
    coordinates: fields['Coordinates'],
    telephone: fields['Phone number'],
    openingTimes: fields['Opening Times'],
    currentProvision: fields['Current Provision'],
    email: fields['Email'],
    referralContact: fields['Referral Contact'],
    selfReferral: fields['Self Referral'],
    notes: fields['Notes'],
    supportingInformation : fields['Supporting information'],
    prompts: fields.Prompt,
    promptTags: fields.Tags
  });
}
