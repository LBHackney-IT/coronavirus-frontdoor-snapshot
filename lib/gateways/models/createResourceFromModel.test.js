import { createResourceFromModel } from './createResourceFromModel';

describe('createResourceFromModel', () => {
  it('maps properties from an Airtable record', () => {
    const record = {
      id: 'dakjhksd8283',
      fields: {
        Name: 'Resource name',
        Description: 'This is a description',
        CategoryDescription: 'Category description',
        Websites: 'https://www.example.com',
        Address: '42 Wallaby Way, Sydney, S1',
        'Post code': 'S1',
        Categories: ['one','two','three'],
        Coordinates: '51.5365,-0.0802',
        Telephone: '077 123456',
        Email: 'a@gmail.com',
        'Referral Email': 'Adam@something',
        'Referral Website': 'Yes.test',
        Notes: 'some notes'
      },
      categoryName: 'one'
    };

    const expectedResponse = {
      address: '42 Wallaby Way, Sydney, S1',
      categoryId: undefined,
      categoryName: 'one',
      coordinates: '51.5365,-0.0802',
      description: 'Category description',
      email: 'a@gmail.com',
      id: 'dakjhksd8283',
      name: 'Resource name',
      postcode: 'S1',
      referralContact: 'Adam@something',
      referralWebsite: 'Yes.test',
      serviceDescription: 'This is a description',
      tags: ['one', 'two', 'three'],
      telephone: '077 123456',
      websites: ['https://www.example.com']
    };

    expect(createResourceFromModel(record)).toMatchObject(expectedResponse);
  });
});
