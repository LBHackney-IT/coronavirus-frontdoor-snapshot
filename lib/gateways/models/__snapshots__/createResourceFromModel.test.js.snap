// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createResourceFromModel maps properties from an Airtable record 1`] = `
Resource {
  "address": "42 Wallaby Way, Sydney, S1",
  "description": "This is a description",
  "id": "dakjhksd8283",
  "name": "Resource name",
  "postcode": "S1",
  "tags": Array [],
  "websites": Array [
    "https://www.example.com",
  ],
}
`;
